game-core/src/test/java/games/strategy/triplea/delegate/battle/MustFightBattleTest.java

```diff
@@ -74,4 +96,293 @@ void testFightWithBothZeroStrength() {
+          Arguments.of(
+              "RetreatInPlace returns battleSite",
+              attacker,
+              Map.of(),
+              List.of(),
+              battleSite,
+              mock(BattleTracker.class),
+              MockGameData.builder().isAtWar(attacker, defender).isRetreatingInPlace().build(),
+              false,
+              Set.of(battleSite)));
+    }
+    {
+      final GamePlayer attacker = mock(GamePlayer.class);
+      final GamePlayer defender = mock(GamePlayer.class);
+      final Territory battleSite =
+          MockTerritory.builder().hasDefenderUnitCollection(defender).build();
+      final Territory attackFrom =
+          MockTerritory.builder().hasEmptyUnitCollection().owner(attacker).build();
+      arguments.add(
+          Arguments.of(
+              "Retreat to attacker's land territory",
+              attacker,
+              Map.of(attackFrom, List.of()),
+              List.of(MockUnit.builder().hasAttachment().build()),
+              battleSite,
+              mock(BattleTracker.class),
+              MockGameData.builder().isAtWar(attacker, defender).landBattleRounds().build(),
+              false,
+              List.of(attackFrom)));
+    }
+    {
+      final GamePlayer attacker = mock(GamePlayer.class);
+      final GamePlayer defender = mock(GamePlayer.class);
+      final Territory battleSite =
+          MockTerritory.builder().hasDefenderUnitCollection(defender).build();
+      final Territory attackFrom =
+          MockTerritory.builder().hasEmptyUnitCollection().nullOwner().isWater().build();
+      arguments.add(
+          Arguments.of(
+              "Retreat to unowned water territory",
+              attacker,
+              Map.of(attackFrom, List.of()),
+              List.of(MockUnit.builder().isSea().build()),
+              battleSite,
+              mock(BattleTracker.class),
+              MockGameData.builder().isAtWar(attacker, defender).landBattleRounds().build(),
+              false,
+              List.of(attackFrom)));
+    }
+    {
+      final GamePlayer attacker = mock(GamePlayer.class);
+      final GamePlayer defender = mock(GamePlayer.class);
+      final Territory battleSite =
+          MockTerritory.builder().hasDefenderUnitCollection(defender).nullOwner().isWater().build();
+      final Territory attackFrom =
+          MockTerritory.builder().hasEmptyUnitCollection().nullOwner().isWater().build();
+      arguments.add(
+          Arguments.of(
+              "Don't include battleSite if it is in the attackedFrom",
+              attacker,
+              Map.of(attackFrom, List.of(), battleSite, List.of()),
+              List.of(MockUnit.builder().isSea().build()),
+              battleSite,
+              mock(BattleTracker.class),
+              MockGameData.builder().isAtWar(attacker, defender).landBattleRounds().build(),
+              false,
+              List.of(attackFrom)));
+    }
+    return arguments;
+  }
+
+  static class MockGameData {
```

Comment:
What do you think of these "mock" builder classes?  Building out the complex objects that are needed for these tests (which are more like an integration test instead of a unit test) can be a little complex with mockito.  Instead of doing a bunch of
```
when(mockObject.function()).thenReturn(something);
when(mockObject.otherFunction()).thenReturn(somethingElse);
```
I'd rather do something more fluent.

But GameData, Unit, Territory, etc do not have a fluent builder api so I can't utilize mockito's fluent apis and even then I'm not sure I'd be able to use them.

So I played around with these builder classes.  It generates a mock object and then gives me methods that I can fluently call to modify the object.  In the end, I have a finished mock that can be used.

I could replace this with a ton of `givenGameData...` methods that deal with the different permutations but that feels like it will become unwieldy with all of the possible permutations.